scripts/import_topo.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2020 ETH Zurich
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+:mod: scripts.import_topo
+=========================
+
+Run with python manage.py runscript import_topo --script-args <topo.yml>
+
+
+Apply topology modifications in batch instead of clicking through the admin panel.
+Only has the features I need to import (many) additional ASes, hosts and links for the GTS/DFN
+"Fed4Fire" setup.
+
+"""
+
+import yaml
+import pathlib
+from django.db import transaction
+from scionlab.models.core import ISD, AS, Host, Service, BorderRouter, Interface, Link
+
+
+class DryRunException(Exception):
+    pass
+
+
+def run(*args):
+    if len(args) != 1:
+        print("run with --script-args <topo-file.yml>")
+        return
+    filename = args[0]
+    topo = yaml.safe_load(pathlib.Path(filename).read_text())
+    dry_run = False
+    try:
+        import_topo(topo, dry_run)
+    except DryRunException:
+        print("Dry run complete, transaction rolled back")
+
+
+@transaction.atomic
+def import_topo(topo, dry_run):
+    ases = topo['ases']
+    links = topo['links']
+    create_ases(ases)
+    create_links(links)
+    if dry_run:
+        raise DryRunException()
+
+
+def create_ases(ases):
+    for ia, data in ases.items():
+        as_ = create_as(ia, data)
+        create_hosts(as_, data['hosts'])
+
+
+def create_as(ia, data):
+    isd_id, as_id = ia.split('-')
+    isd = ISD.objects.filter(isd_id=isd_id).get()
+
+    is_core = data.get('core', False)
+    label = data['label']
+    replace = data.get('replace', True)
```

Comment:
I'd have 'replace' being False by default, seems safer/less prone to accidentally removing hosts
Since most of the ASes that are imported here do not exist, it doesn't make much of a difference

Reply:
Done.